metpy/io/upperair.py

```diff
@@ -167,15 +167,54 @@ def parse(fobj):
             fobj.readline()
 
         # Parse the actual data, only grabbing the columns for pressure, T/Td, and wind
-        cols = (0, 2, 3, 6, 7)
         unit_strs = ['degC' if u == 'C' else u
                      for u in fobj.readline().decode('ascii').split()]
 
-        # Skip 2 header lines -- 1 for '----' and 1 for partial line at 1000mb
-        p, t, td, direc, spd = np.genfromtxt(fobj, usecols=cols, skip_header=2, unpack=True)
-
-        return dict(p=(p, unit_strs[0]), t=(t, unit_strs[2]), td=(td, unit_strs[3]),
-                    wind=(direc, spd, unit_strs[7]))
+        # Parse to the actually beginning of data at or above ground.
+        fobj.readline()
+        flag = False
+        for _ in range(5):
+            x = fobj.tell()
+            ind = fobj.readline().decode('ascii').split()
+            if (len(ind) == 11) & (not flag):
+                dx = x
+                flag = True
+        fobj.seek(dx)
+
+        # Initiate lists for variables
+        p = []
+        t = []
+        td = []
+        direc = []
+        spd = []
+
+        # Read all lines of data and append to lists
+        data = fobj.readlines()
+        for rows in range(len(data)):
+            p.append(data[rows][0:7])
+            t.append(data[rows][14:21])
+            td.append(data[rows][21:28])
```

Comment:
Could call the local function here:
```python
arr_data.append((to_float(data[rows][0:7]), to_float(data[rows][14:21]), ...)
```
Note I'm appending a tuple of multiple values here rather than one...